add `with_addons` param to Collection detail/view API

eviljeff opened the pull request:
fixes #6548 - could be more complete and have full pagination and sorting options for the addons, but this seems sufficient... ?

diox commented:
Is this still needed on the frontend ? I fear this might just trade perf on the server for perf on the frontend...

kumar303 commented on docs/topics/api/collections.rst:
```diff
@@ -58,6 +65,11 @@ If your account has the `Collections:Edit` permission then you can access any co
     :>json string uuid: A unique identifier for this collection; primarily used to count addon installations that come via this collection.
 
 
+If the ``with_addon`` parameter is passed then :ref:`addons in the collection<collection-addon-detail>` are returned along with the detail.
```
Could you document how that looks in the response? I assume it will just add an `addons` key to the detail object? The pattern the docs usually follow is something like:

Response JSON Object:

- **id**: ...
- **addon_count**: ...
- **addons**: (null|array) Some description here and a link to the add-on object.

kumar303 commented on docs/topics/api/collections.rst:
```diff
@@ -27,6 +27,11 @@ The results are sorted by the most recently updated collection first.
     :>json array results: An array of :ref:`collections <collection-detail-object>`.
 
 
+If the ``with_addon`` parameter is passed then :ref:`addons in the collection<collection-addon-detail>` are returned along with the detail.
```
In case this helps, I can't think of any situation where we will need addons as part of the result returned from the list API. We do need addons from the detail API.

eviljeff replied:
It's easier _not_ to check if it's a detail or list request, tbh, but I could limit this to detail responses to reduce the server load.

kumar303 commented:
> Is this still needed on the frontend ? I fear this might just trade perf on the server for perf on the frontend...

Currently, we always have to make two requests (one for metadata and one for add-ons) when the user views a collection. With this new patch we could remove one API request. There's no visible performance problem with those two requests but each HTTP request has a lot of roundtrips. So, yeah, it's not currently broken and it's not "slow." If you don't think it's worth it to make this change on the server then no biggie.

eviljeff commented:
addressed the requested changes and also refactored to use the same code that Collection Addon uses (so we get consistency around what's returned vs. the full endpoint) - with the bonus that the filter param is respected.  https://github.com/mozilla/addons-server/pull/7546/commits/05aea04d5a38386f3aa9ef5d330a9456ed877fbd

diox commented on src/olympia/bandwagon/serializers.py:
```diff
@@ -109,3 +109,18 @@ def validate(self, data):
+    def get_addons(self, obj):
+        addons_qs = self.context['view'].get_addons_queryset()
+        return [
+            CollectionAddonSerializer(addon).data for addon in addons_qs
```
You should be able to just do `return CollectionAddonSerializer(addons_qs, many=True).data` instead.

In fact, it may even work with `addons = CollectionAddonSerializer(source='fetch_addons', many=True)` above and just `return self.context['view'].get_addons_queryset()` in a `fetch_addons` method. 

Any of those 2 approaches should work (hopefully, haven't tested) and avoids creating extra serializer instances, which is a bit costly.

eviljeff replied:
latter didn't work bit did the former https://github.com/mozilla/addons-server/pull/7546/commits/02830c9e8bcbb34e725ea106f697b18951f2f730
